[CI] XPackRestIT test {yaml=search.vectors/47_knn_search_bbq_disk_slice/KNN search with a different _slice returns that slice's nearest neighbor} failing

**Build Scans:**
- [elasticsearch-periodic #12412 / release-tests](https://gradle-enterprise.elastic.co/s/4avrpkh6gdo6m)
- [elasticsearch-periodic #12412 / release-tests](https://gradle-enterprise.elastic.co/s/5hddv6d2mxy7c)
- [elasticsearch-pull-request #155042 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/3ivttssbmijoe)
- [elasticsearch-pull-request #155042 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/lpjrfjwvs77o6)
- [elasticsearch-pull-request #154992 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/fsw4y553hisx2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT" -Dtests.method="test {yaml=search.vectors/47_knn_search_bbq_disk_slice/KNN search with a different _slice returns that slice's nearest neighbor}" -Dtests.seed=62FEBDFE5BA8A6C9 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=zh-TW -Dtests.timezone=Asia/Harbin -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Byaml%3Dsearch.vectors%2F47_knn_search_bbq_disk_slice%2FKNN%20search%20with%20a%20different%20_slice%20returns%20that%20slice's%20nearest%20neighbor%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: org.junit.internal.runners.model.MultipleFailureException: There were 3 errors:
  java.lang.AssertionError(field [hits.total.value] does not match the expected value
Expected: <2>
     but: was <1>)
  java.lang.AssertionError(field [hits.hits] doesn't have length [2]
Expected: <2>
     but: was <1>)
  java.lang.AssertionError(Error executing section at [search.vectors/47_knn_search_bbq_disk_slice:92]: element was a list with 1 elements, but [1] was out of bounds)
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {yaml=search.vectors/47_knn_search_bbq_disk_slice/KNN search with a different _slice returns that slice's nearest neighbor}
- [main] 2 consecutive failures in step release-tests
- [main] 3 consecutive failures in step checkpart1_release-tests
- [main] 1 consecutive failures in pipeline elasticsearch-periodic
- [main] 5 failures in test test {yaml=search.vectors/47_knn_search_bbq_disk_slice/KNN search with a different _slice returns that slice's nearest neighbor} (2.6% fail rate in 194 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)
- [main] 3 failures in step checkpart1_release-tests (100.0% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.5% fail rate in 137 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -641,9 +641,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.datafeed.DatafeedManagerTests
   method: testUpdateDatafeedFirstTimeMigrationShouldNotOverrideExistingProjectRouting
   issue: https://github.com/elastic/elasticsearch/issues/151702
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {yaml=search.vectors/47_knn_search_bbq_disk_slice/KNN search with a different _slice returns that slice's nearest neighbor}
-  issue: https://github.com/elastic/elasticsearch/issues/151703
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:string.mvAppendNull}
   issue: https://github.com/elastic/elasticsearch/issues/151707

--- a/x-pack/plugin/build.gradle
+++ b/x-pack/plugin/build.gradle
@@ -55,12 +55,13 @@ if (buildParams.snapshotBuild == false) {
   // Since there is no infrastructure in place (anytime soon) to generate licenses using the production
   // private key, these tests are blacklisted in non-snapshot test runs
   restTestBlacklist.addAll(['xpack/15_basic/*', 'license/20_put_license/*', 'license/30_enterprise_license/*'])
-
   // TODO: Remove the following when the following features are released. These tests include new privileges only available under feature flags
   //  which require snapshot builds:
   // * Data Stream Lifecycle. manage_data_stream_lifecycle privilege is only available with dlm_feature_flag_enabled set
   // We disable these tests for snapshot builds to maintain release build coverage.
   restTestBlacklist.add('privileges/11_builtin/Test get builtin privileges')
+  // Requires ESNext codec which is not available in release test. This should be removed once the production codec supports slices.
+  restTestBlacklist.add('search.vectors/47_knn_search_bbq_disk_slice/*')
 }
 
 tasks.withType(StandaloneRestIntegTestTask).configureEach {

